File: pjecalc-fonte/negocio/br/jus/trt8/pjecalc/negocio/dominio/calculo/relatorio/PrevidenciaPrivadaJRAdapter.java
/*
 * Decompiled with CFR 0.152.
 */
package br.jus.trt8.pjecalc.negocio.dominio.calculo.relatorio;

import br.jus.trt8.pjecalc.base.comum.relatorio.JRAdapter;
import br.jus.trt8.pjecalc.base.comum.relatorio.JRAdapterDataSource;
import java.math.BigDecimal;
import java.util.Date;

public abstract class PrevidenciaPrivadaJRAdapter
extends JRAdapter {
    public abstract String getFormula();

    public abstract BigDecimal getTotalDevido();

    public abstract JRAdapterDataSource<PrevidenciaPrivadaOcorrenciaAdapter> getOcorrencias();

    public abstract BigDecimal getTotalDoDevidoCorrigido();

    public abstract BigDecimal getTotalDeJuros();

    public abstract BigDecimal getTotalGeral();

    public abstract boolean getMostrarJuros();

    public abstract class PrevidenciaPrivadaOcorrenciaAdapter
    extends JRAdapter {
        public abstract Date getOcorrencia();

        public abstract BigDecimal getBase();

        public abstract BigDecimal getAliquota();

        public abstract BigDecimal getDevido();

        public abstract BigDecimal getIndiceCorrecao();

        public abstract BigDecimal getDevidoCorrigido();

        public abstract BigDecimal getJuros();

        public abstract BigDecimal getTotal();
    }
}

File: src/lib/pjecalc/core/dominio/calculo/relatorio/previdencia-privada-jr-adapter.ts
/**
 * PJe-Calc v2.15.1 — PrevidenciaPrivadaJRAdapter (abstract)
 * Porte TS-adaptado de:
 *   br.jus.trt8.pjecalc.negocio.dominio.calculo.relatorio.PrevidenciaPrivadaJRAdapter
 *
 * Ref Java: pjecalc-fonte/.../relatorio/PrevidenciaPrivadaJRAdapter.java
 */
import type Decimal from 'decimal.js';
import { JRAdapter, JRAdapterDataSource } from './jr-adapter';

export abstract class PrevidenciaPrivadaJRAdapter extends JRAdapter {
  abstract getOcorrencias(): JRAdapterDataSource<PrevidenciaPrivadaOcorrenciaAdapter>;
  abstract getTotalDevido(): Decimal;
  abstract getTotalDevidoCorrigido(): Decimal;
}

export abstract class PrevidenciaPrivadaOcorrenciaAdapter extends JRAdapter {
  abstract getCompetencia(): Date;
  abstract getBase(): Decimal;
  abstract getAliquota(): Decimal;
  abstract getValor(): Decimal;
  abstract getIndiceCorrecao(): Decimal;
  abstract getValorCorrigido(): Decimal;
}
